### PR TITLE
Gestures

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### What is the current behavior?

A Gaphas view has it's own handling of events, through tools.

Issue Number: #72

### What is the new behavior?
Tools are a lot like Gtk's EventControllers. Those will be used instead.

### Does this PR introduce a breaking change?
- [X] Yes
- [ ] No

### Other information

* Requires Gtk 3.24 or newer
* To make use of EventControllers, I needed up upgrade the build servers to Ubuntu 20.04.
* `EventControllers` are created with a widget reference.

I edited a bit of documentation. It will probably require something of a rewrite though. documentation updates I want to do in a later PR.

### CODE REVIEW

File: gaphas/tool/hover.py [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
-        pos = event.get_coords()[1:]
-        view.selection.set_hovered_item(Finder(view).get_item_at_point(pos))
+def hover_tool(view: GtkView):
+    ctrl = Gtk.EventControllerMotion.new(view)
+    ctrl.connect("motion", on_motion)
+    return ctrl
+
+
+def on_motion(ctrl, x, y):
+    # TODO: Check for claimed controller
```
Comment (danyeaw) [inline gaphas/tool/hover.py:15]: Do we need to do anything with this TODO before merging?

File: gaphas/tool/itemtool.py [hunk 2]
```diff
@@ -20,91 +25,160 @@ def start_move(self, pos: Pos):
...
+    for moving in drag_state.moving:
+        moving.stop_move((x + offset_x, y + offset_y))
+    drag_state.moving = set()
+
+
+def item_at_point(view: GtkView, pos: Pos, selected=True) -> Optional[Item]:
+    """Return the topmost item located at ``pos`` (x, y).
+
+    Parameters:
+        - selected: if False returns first non-selected item
```
Comment (danyeaw) [inline gaphas/tool/itemtool.py:120]: This docstring is missing parameters.

**Review by danyeaw — APPROVED** [review]
This is so awesome to upgrade Gaphas to use Gtk Gestures, really nice job @amolenaar! :tada:

File: gaphas/tool/hover.py (hunk 1 above)
Comment (amolenaar) [reply gaphas/tool/hover.py:15]: I think this todo is already resolved.

**Pull request merged**